### DISCUSSION TITLE
Symfony : send JsonResponse

### DISCUSSION THREAD

**Discussion by Willy2kid** [community/community#113766] [Programming Help]
### Body

Hello, i'm new to symfony as web programing and i'm having a hard time with the use of json response.

As you can see, the controller display the entity content and create a form to add data in it. The form work fine and the array send by ajax is added to the entity BUT the idea is to send back this data as a json response to update the entity content without refreshing the whole page.

Problem is that when the form is send, the json response replace the page and show something like :
{
    "success": true,
    "new_object": {
        "id": 27,
        "name": "thename",
        "description": "thedescription",
        "date": "2024-03-23T10:22:00+00:00",
    }
}

I've thought that i should replace Response by JsonResponse when the function is defined but the the render of the view send an error asking for a json response. 

I also fought to create a new function to send the json response but because $form is handling the request i can't see how to add the form to the page is separate in another function.

How should i send this json response then ? Any help is really welcome 👍 

`<?php

namespace App\Controller;

use App\Entity\MyEntity;
use App\Form\OffresFormType;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;
use Doctrine\ORM\EntityManagerInterface;
use Symfony\Component\HttpFoundation\JsonResponse;

class MyEntityController extends AbstractController
{
    /**
     * @Route("/admin", name="administration", methods={"GET", "POST"})
     */
    public function admin(Request $request, EntityManagerInterface $entityManager): Response
    {

        $newobject = new MyEntity();
        $form = $this->createForm(MyEntityFormType::class, $newobject);
        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {
            $entityManager->persist($newobject);
            $entityManager->flush();

            return $this->json(['success' => true, 'newobject' => $newobject]);
        }

        return $this->render('admin/entity.html.twig', [
            'entity_content' => $entityManager->getRepository(MyEntity::class)->findAll(),
            'form' => $form->createView(),
        ]);
    }
}
`

**Comment by liker912** [comment]
Hi @Willy2kid , and welcome. I think for the POST request you need to create another method (e.g. admin_attempt) that will validate a form and will return a JSON only

````    /**
     * @Route("/admin", name="admin_index")
     */
    public function admin(EntityManagerInterface $entityManager): Response
    {
        $entities = $entityManager->getRepository(MyEntity::class)->findAll();

        return $this->render('admin/entity.html.twig', [
            'entity_content' => $entities,
            'form' => $this->createForm(MyEntityFormType::class)->createView(),
        ]);
    }

    /**
     * @Route("/admin/save", name="admin_save", methods={"POST"})
     */
    public function admin_atemt(Request $request, EntityManagerInterface $entityManager): Response
    {
        $newobject = new MyEntity();
        $form = $this->createForm(MyEntityFormType::class, $newobject);
        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {
            $entityManager->persist($newobject);
            $entityManager->flush();

            return $this->json(['success' => true, 'newobject' => $newobject]);
        }

        return $this->render('admin/entity.html.twig', [
            'entity_content' => $entityManager->getRepository(MyEntity::class)->findAll(),
            'form' => $form->createView(),
        ]);
    }
}
